solr/solrj/src/java/org/apache/solr/client/solrj/io/ClassificationEvaluation.java

```diff
@@ -36,28 +36,26 @@ public void count(int actual, int predicted) {
     }
   }
 
-  @SuppressWarnings({"unchecked"})
-  public void putToMap(@SuppressWarnings({"rawtypes"})Map map) {
+  public void putToMap(Map<? super String, ? super Long> map) {
     map.put("truePositive_i",truePositive);
     map.put("trueNegative_i",trueNegative);
     map.put("falsePositive_i",falsePositive);
     map.put("falseNegative_i",falseNegative);
   }
 
-  @SuppressWarnings({"rawtypes"})
-  public Map toMap() {
-    HashMap map = new HashMap();
+  public Map<String, Long> toMap() {
+    HashMap<String, Long> map = new HashMap<>();
     putToMap(map);
     return map;
   }
 
-  public static ClassificationEvaluation create(@SuppressWarnings({"rawtypes"})Map map) {
+  public static ClassificationEvaluation create(Map<String, ?> map) {
     ClassificationEvaluation evaluation = new ClassificationEvaluation();
     evaluation.addEvaluation(map);
     return evaluation;
   }
 
-  public void addEvaluation(@SuppressWarnings({"rawtypes"})Map map) {
+  public void addEvaluation(Map<String, ?> map) {
```

Comment:
```suggestion
  public void addEvaluation(Map<String, ? super Long> map) {
```

Reply:
Following up on the earlier comment about PECS, this map is a producer, not a consumer, so we don't care about making sure that it can accept Long. We're kind of dancing on a sharp edge here where we cast things to longs from ?, but we don't know what's in the Map and we're not putting anything new in it so we're ok. If there's an assumption violated about data types here, then we'll throw an exception and that's probably the best we can do regardless.